test/AutoDiff/generics.swift

```diff
@@ -52,7 +52,7 @@ struct SupervisedTrainer<Model : Layer> {
   var model: Model
   var lossFunction: @differentiable (Model.Output, Model.Output) -> Float
   func fit(y: Model.Output) {
-    _ = gradient(at: y) { y in return lossFunction(y, y) }
+    _ = gradient(at: y) { y in return self.lossFunction(y, y) }
```

Comment:
Is adding `self.` necessary for some reason?

Reply:
Yes. `self` is being captured in an escaping closure first before the closure gets converted to `@noescape`, and thus needs to be explicit.